packages/enzyme-test-suite/test/Adapter-spec.jsx

```diff
@@ -1010,6 +1013,56 @@ describe('Adapter', () => {
     itIf(is('>= 16.6'), 'supports ConcurrentMode', () => {
       expect(getDisplayName(<ConcurrentMode />)).to.equal('ConcurrentMode');
     });
+
+    itIf(is('>= 16.6'), 'supports Suspense', () => {
+      expect(getDisplayName(<Suspense />)).to.equal('Suspense');
+    });
+
+    itIf(is('>= 16.6'), 'supports lazy', () => {
+      class DynamicComponent extends React.Component {
+        render() {
+          return <div>DynamicComponent</div>;
+        }
+      }
+      const LazyComponent = lazy(() => fakeDynamicImport(DynamicComponent));
+      expect(getDisplayName(<LazyComponent />)).to.equal('lazy');
+    });
+
+    itIf(is('>= 16.6'), 'show explicitly defined display name of lazy component', () => {
+      class DynamicComponent extends React.Component {
+        render() {
+          return <div>DynamicComponent</div>;
+        }
+      }
+      const theDisplayName = 'SOMETHING';
+      const LazyComponent = Object.assign(lazy(() => fakeDynamicImport(DynamicComponent)), { displayName: theDisplayName });
+      expect(getDisplayName(<LazyComponent />)).to.equal(theDisplayName);
+    });
+
+    itIf(is('>= 16.6'), 'show display name of wrapped component of lazy', () => {
+      class ComponentWithDisplayName extends React.Component {
+        render() {
+          return <div>DynamicComponent</div>;
+        }
+      }
+      ComponentWithDisplayName.displayName = 'Something';
+      const LazyComponent = lazy(() => fakeDynamicImport(ComponentWithDisplayName));
+      /* eslint-disable no-underscore-dangle */
+      LazyComponent._result = ComponentWithDisplayName;
```

Comment:
why is this needed?

Reply:
I'd like to test what displayName will be if the wrapped component is loaded. And in lazy component we could get the wrapped component in `_result`.

Reply:
I'd prefer not to include these implementation details in the test. Can it be tested by providing an actually loaded component, or by awaiting the promise, rather than this?